### PR TITLE
Add new modules

### PR DESCRIPTION
**Pull request by daichengxin**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/quantms/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/quantms _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/local/decoydatabase/main.nf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    publishDir "${params.outdir}/logs",
+        mode: params.publish_dir_mode,
+        pattern: '*.log',
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:[:], publish_by_meta:[]) }
+
+    conda (params.enable_conda ? "openms::openms-thirdparty=2.7.0pre" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/openms:2.6.0--h4afb90d_0"
+    } else {
+        container "quay.io/biocontainers/openms:2.6.0--h4afb90d_0"
```
Comment (ypriverol) [inline modules/local/decoydatabase/main.nf:18]: @daichengxin we need to create an issue that make clears that all these conda should be updated when the dev of 2.7.0 is available.

File: modules/local/decoydatabase/meta.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    - log:
+        type: file
+        description: log file
+        pattern: "*.log"
+    - version:
+        type: file
+        description: File containing software version
+        pattern: "*.{version.txt}"
+authors:
+    - "@Chengxin Dai"
```
Comment (ypriverol) [inline modules/local/decoydatabase/meta.yml:32]: What is the still of the authors: email? github account?.

File: modules/local/fileconverter/main.nf [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process FILECONVERTER {
```
Comment (ypriverol) [inline modules/local/fileconverter/main.nf:7]: I think this is more like mzML indexing. I suggest the following name: MZMLINDEXING    Opinions @jpfeuffer

File: modules/local/sdrfparsing/main.nf [hunk 4]
```diff
@@ -0,0 +1,40 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process SDRFPARSING {
```
Comment (ypriverol) [inline modules/local/sdrfparsing/main.nf:7]: SDRFTOOPENMS

Comment (ypriverol) [reply modules/local/sdrfparsing/main.nf:7]: @daichengxin if the DIA pipeline will not convert to open MS, then is probably better to keep SDRFPARSING

**Review by ypriverol — CHANGES_REQUESTED** [review]

File: modules/local/decoydatabase/main.nf (hunk 1 above)
Comment (jpfeuffer) [reply modules/local/decoydatabase/main.nf:18]: Can we put all openms modules in a subfolder?
openms/$nameofthetool

Maybe even another one for third-party tools

openms/thirdparty/$nameoftheadapter

I bet we can then define shared variables for all those modules in the parent folder

File: modules/local/fileconverter/main.nf (hunk 3 above)
Comment (jpfeuffer) [reply modules/local/fileconverter/main.nf:7]: Hmm I could see both sides.
If we generalize it enought I think we should leave it as Fileconverter.
We could then write a wrapper/module/subworkflow around it, that calls it in a way that it only does indexing. And call that step mzmlindexing then.

File: modules/local/sdrfparsing/main.nf (hunk 4 above)
Comment (jpfeuffer) [reply modules/local/sdrfparsing/main.nf:7]: Both is fine for me. Maybe at some point people want to do sdrf to maxquant.

**Comment by jpfeuffer** [comment]
Wow that was quick. Great work. I will need some time to go over everything.

File: modules/local/fileconverter/main.nf (hunk 3 above)
Comment (ypriverol) [reply modules/local/fileconverter/main.nf:7]: I fear is that I don't want to "overcode" if the tool supports more functionalities we should not the module as the tool, we should create modules for functionalities that are in use in the process.

File: modules/local/decoydatabase/main.nf (hunk 1 above)
Comment (ypriverol) [reply modules/local/decoydatabase/main.nf:18]: Can you send and example @jpfeuffer about how to do shared variables.

Comment (daichengxin) [reply modules/local/decoydatabase/main.nf:18]: ok, I will create an issue to illustrate version. Putting all openms modules in a subfolder is a good idea. For shared variables, I think we can define them in modules.config file

File: modules/local/decoydatabase/meta.yml (hunk 2 above)
Comment (daichengxin) [reply modules/local/decoydatabase/meta.yml:32]: github accout

**Comment by daichengxin** [comment]
I am also learning DSL2, and many codes may be optimized and corrected.

File: modules/local/decoydatabase/meta.yml (hunk 2 above)
Comment (ypriverol) [reply modules/local/decoydatabase/meta.yml:32]: If is the GitHub account, your account is @daichengxin 😃

Comment (daichengxin) [reply modules/local/decoydatabase/meta.yml:32]: Got it! @ypriverol

**Review by ypriverol — APPROVED** [review]

**Pull request merged**